### PR TITLE
CDI Events for Consumer/Producer

### PR DESCRIPTION
**Pull request by ebullient**
Skipping qualifier for create/delete for now.. so the event is only for create. 
Some other detection (or maybe a separate event entirely) would be needed for destroy.

### CODE REVIEW

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    public Event<Producer<?, ?>> producer() {
+        return producerEvent;
+    }
+
+    public KafkaCDIEvents() {
+    }
+
+    private KafkaCDIEvents(boolean isNoop) {
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java:34]: Can you add a comment about this constructor that is not used during "normal" operations.

Comment (cescoffier) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java:34]: Is it even used?

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+import javax.enterprise.event.Event;
+import javax.enterprise.event.NotificationOptions;
+import javax.enterprise.util.TypeLiteral;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.Producer;
+
+public class CountKafkaCdiEvents extends KafkaCDIEvents {
+    public static final KafkaCDIEvents noCdiEvents = new CountKafkaCdiEvents();
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java:15]: Should we reset the counter at the end of each test?

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+    public final LongAdder firedConsumerEvents = new LongAdder();
```
Comment (cescoffier) [inline smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java:17]: Nice, I always used AtomicLong! @jponge might be interesting for the spies.

**Review by cescoffier — CHANGES_REQUESTED** [review]
That's looks great! 

Two comments:

* not sure the private constructor in the KafkaCDIEvents class is used.
* wondering if we need to add a line in the doc, or if we should keep this under the radar until the metrics side is implemented.

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java (hunk 3 above)
Comment (jponge) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java:17]: Yep

**Comment by ebullient** [comment]
Yea.. I thought I removed it? May not have pushed.

File: smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java (hunk 1 above)
Comment (ebullient) [reply smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java:34]: Not anymore. I thought I removed it.. but I may not have pushed the change. Will double check.

File: smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java (hunk 2 above)
Comment (ebullient) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java:15]: For this one.. it's saying "I don't care" .. so I didn't bother resetting the counter on this one.

Comment (cescoffier) [reply smallrye-reactive-messaging-kafka/src/test/java/io/smallrye/reactive/messaging/kafka/CountKafkaCdiEvents.java:15]: ok

**Review by cescoffier — APPROVED** [review]

**Pull request merged**